### PR TITLE
Add geo_in_rectangle method in `select.rb`

### PR DESCRIPTION
**Pull request by komainu8**

### CODE REVIEW

**Comment by kou** [comment]
I've refectored and your changes are out of date.
Can you rebase (not merge) your change on master? Or should I do by myself?

**Comment by komainu8** [comment]
I'll do it by myself. Thanks.

**Comment by komainu8** [comment]
I fixed confricts in my branch.

File: lib/groonga/client/request/select.rb [hunk 1]
```diff
@@ -93,6 +93,11 @@ def query(value)
         #        filter.in_values(:tags, "tag1", "tag2")
         #          # -> --filter 'in_values(tags, "tag1", "tag2")'
         #
+        #   @example: Use geo_in_rectangle function
+        #      request.
+        #        filter.geo_in_rectangle("50x50", "0x100", "100x0")
+        #          # -> --filter 'geo_in_rectangle("50x50", "0x100", "100x0")'
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: Normally, the first argument is column.

File: lib/groonga/client/request/select.rb [hunk 2]
```diff
@@ -218,6 +223,39 @@ def initialize(request)
...
+          # Adds a `geo_in_rectangle` condition then return a new `select`
+          # request object.
+          #
+          # @see http://groonga.org/docs/reference/functions/geo_in_rectangle.html
+          #   geo_in_rectangle function in the Groonga document
+          #
+          # @example: Basic usage
+          #    request.
+          #      filter.geo_in_rectangle("50x50", "0x100", "100x0").
+          #        # -> --filter 'in_values(50x50, 0x100, 100x0)'
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: ditto.

**Comment by komainu8** [comment]
Thank you very much for your comments. I've fixed it.

File: lib/groonga/client/request/select.rb [hunk 3]
```diff
@@ -218,6 +223,41 @@ def initialize(request)
...
+          # Adds a `geo_in_rectangle` condition then return a new `select`
+          # request object.
+          #
+          # @see http://groonga.org/docs/reference/functions/geo_in_rectangle.html
+          #   geo_in_rectangle function in the Groonga document
+          #
+          # @example: Basic usage
+          #    request.
+          #      filter.geo_in_rectangle(:location, "0x100", "100x0").
+          #        # -> --filter 'in_values(location, 0x100, 100x0)'
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: Result is wrong.

File: lib/groonga/client/request/select.rb [hunk 4]
```diff
@@ -218,6 +223,41 @@ def initialize(request)
...
+          #    `"#{LONGITUDE}x#{LATITUDE}"` is the point format.
+          #
+          # @param bottom_right [String] The bottom right of the condition rectangle.
+          #    `"#{LONGITUDE}x#{LATITUDE}"` is the point format.
+          #
+          # @return [Groonga::Client::Request::Select]
+          #   The new request with the given condition.
+          #
+          # @since 0.5.0
+          def geo_in_rectangle(column_name_or_point,
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: Document for the first column is `point` case is missing.

**Comment by komainu8** [comment]
Thank you for your comments. I've fixed documents of geo_in_rectangle.

File: lib/groonga/client/request/select.rb [hunk 5]
```diff
@@ -218,6 +223,59 @@ def initialize(request)
...
+          #
+          # @see http://groonga.org/docs/reference/functions/geo_in_rectangle.html
+          #   geo_in_rectangle function in the Groonga document
+          #
+          # @overload geo_in_rectangle(column_name, top_left, bottom_right)
+          #
+          #   @example: Basic usage
+          #      request.
+          #        filter.geo_in_rectangle(:location, "0x100", "100x0").
+          #          # -> --filter 'geo_in_rectangle(location, 0x100, 100x0)'
```
Comment (kou) [inline lib/groonga/client/request/select.rb]: The result is wrong.

**Comment by komainu8** [comment]
Thank you for your comment. I've fixed example of geo_in_rectangle.

**Pull request merged**